patients/deletions_repo.go

```diff
@@ -0,0 +1,74 @@
+package patients
+
+import (
+	"context"
+	"fmt"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"go.uber.org/fx"
+	"go.uber.org/zap"
+)
+
+const (
+	PatientDeletionsCollectionName = "patient_deletions"
+)
+
+//go:generate mockgen --build_flags=--mod=mod -source=./deleted_repo.go -destination=./test/mock_deleted_repository.go -package test -aux_files=github.com/tidepool-org/clinic/patients=patients.go MockDeletedRepository
+
+type PatientDeletionsRepository interface {
+	Create(context.Context, PatientDeletion) error
+}
+
+
+func NewPatientDeletionsRepository(db *mongo.Database, logger *zap.SugaredLogger, lifecycle fx.Lifecycle) (PatientDeletionsRepository, error) {
+	repo := &patientDeletionsRepository{
+		collection: db.Collection(PatientDeletionsCollectionName),
+		logger:     logger,
+	}
+
+	lifecycle.Append(fx.Hook{
+		OnStart: func(ctx context.Context) error {
+			return repo.Initialize(ctx)
+		},
+	})
+
+	return repo, nil
+}
+
+type patientDeletionsRepository struct {
+	collection *mongo.Collection
+	logger     *zap.SugaredLogger
+}
+
+func (p *patientDeletionsRepository) Initialize(ctx context.Context) error {
+	_, err := p.collection.Indexes().CreateMany(ctx, []mongo.IndexModel{
+		{
+			Keys: bson.D{
+				{Key: "patient.clinicId", Value: 1},
+				{Key: "patient.userId", Value: 1},
+			},
+			Options: options.Index().
+				SetName("PatientDeletion"),
+		},
+		{
+			Keys: bson.D{
+				{Key: "deletedTime", Value: 1},
+				{Key: "patient.clinicId", Value: 1},
+			},
+			Options: options.Index().
+				SetName("DeletedTime"),
+		},
+	})
+	return err
+}
+
+func (p *patientDeletionsRepository) Create(ctx context.Context, deletion PatientDeletion) error {
+	if deletion.DeletedTime.IsZero() {
```

Comment:
Do we want to check that the patient id is not nil?

Reply:
Being more defensive won't hurt